heka/files/lua/encoders/status_nagios.lua

```diff
@@ -64,10 +66,17 @@ function process_message()
         return -1
     end
 
+    local host
     if host_dimension_key then
-        data['host'] = read_message(string.format('Fields[%s]', host_dimension_key)) or default_host
+        host = read_message(string.format('Fields[%s]', host_dimension_key)) or default_host
+    else
+        host = read_message('Fields[hostname]') or read_message('Hostname')
+    end
+
+    if host_suffix_dimension_key then
```

Comment:
(nit) I would read `Fields[<host_suffix_dimension_key>]` first and only append if it's not nil.